HPXMLOutputReport/measure.rb

```diff
@@ -0,0 +1,418 @@
+    end
+
+    output_format = runner.getStringArgumentValue('output_format', user_arguments)
+
+    sqlFile = runner.lastEnergyPlusSqlFile
+    if sqlFile.empty?
+      runner.registerError('Cannot find EnergyPlus sql file.')
+      return false
+    end
+    sqlFile = sqlFile.get
+    if not sqlFile.connectionOpen
+      runner.registerError('EnergyPlus simulation failed.')
+      return false
+    end
+    model.setSqlFile(sqlFile)
+
+    hpxml_defaults_path = model.getBuilding.additionalProperties.getFeatureAsString('hpxml_defaults_path').get
+    hpxml = HPXML.new(hpxml_path: hpxml_defaults_path)
+
+    # Set paths
+    output_dir = File.dirname(sqlFile.path.to_s)
+    output_path = File.join(output_dir, "hpxml_output.#{output_format}")
+
+    sqlFile.close()
+
+    # Ensure sql file is immediately freed; otherwise we can get
+    # errors on Windows when trying to delete this file.
+    GC.start()
+
+    # Initialize
+    cost_multipliers = {}
+    cost_multipliers[BS::Fixed] = BaseOutput.new
+    cost_multipliers[BS::WallAreaAboveGradeConditioned] = BaseOutput.new
+    cost_multipliers[BS::WallAreaAboveGradeExterior] = BaseOutput.new
+    cost_multipliers[BS::WallAreaBelowGrade] = BaseOutput.new
+    cost_multipliers[BS::FloorAreaConditioned] = BaseOutput.new
+    cost_multipliers[BS::FloorAreaAttic] = BaseOutput.new
+    cost_multipliers[BS::FloorAreaLighting] = BaseOutput.new
+    cost_multipliers[BS::RoofArea] = BaseOutput.new
+    cost_multipliers[BS::WindowArea] = BaseOutput.new
+    cost_multipliers[BS::DoorArea] = BaseOutput.new
+    cost_multipliers[BS::DuctUnconditionedSurfaceArea] = BaseOutput.new
+    cost_multipliers[BS::SizeWaterHeater] = BaseOutput.new
+    cost_multipliers[BS::FlowRateMechanicalVentilation] = BaseOutput.new
+    cost_multipliers[BS::SlabPerimeterExposedConditioned] = BaseOutput.new
+    cost_multipliers[BS::RimJoistAreaAboveGradeExterior] = BaseOutput.new
+
+    hpxml.heating_systems.each do |heating_system|
+      cost_multipliers["#{BS::SizeHeatingSystem}: #{heating_system.id}"] = BaseOutput.new
+    end
+
+    hpxml.cooling_systems.each do |cooling_system|
+      cost_multipliers["#{BS::SizeCoolingSystem}: #{cooling_system.id}"] = BaseOutput.new
+    end
+
+    hpxml.heat_pumps.each do |heat_pump|
+      cost_multipliers["#{BS::SizeHeatingSystem}: #{heat_pump.id}"] = BaseOutput.new
+      cost_multipliers["#{BS::SizeCoolingSystem}: #{heat_pump.id}"] = BaseOutput.new
+      cost_multipliers["#{BS::SizeHeatPumpBackup}: #{heat_pump.id}"] = BaseOutput.new
+    end
+
+    # Cost multipliers
+    cost_multipliers.each do |cost_mult_type, cost_mult|
+      cost_mult.output = get_cost_multiplier(hpxml, cost_mult_type)
+    end
+
+    # Primary and Secondary
+    if hpxml.primary_hvac_systems.size > 0
+      assign_primary_and_secondary(hpxml, cost_multipliers)
+    end
+
+    # Units
+    cost_multipliers.each do |cost_mult_type, cost_mult|
+      cost_mult.name = "Building Summary: #{cost_mult_type}"
```

Comment:
Rather than prefix every row with "Building Summary", I suggest segmenting by:
- "Surface Area: " (you can then remove "area" from the individual names)
- "Sizes: " (you can then remove "size" from the individual names
- "Other: " (mech vent flow rate and slab perimeter)

And please add blank lines between the sections (like we do in the results_annual.csv output).